.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+name: Tests
+
+on: [push, pull_request]
```

Comment:
I _think_ that by defining the `on` this way we may have unwanted effects:
- it produces double triggers on pushes to branches that have a corresponding PR. This is because it triggers:
  - when there is a `push` to any given branch (stemming from the `on: push`)
  - when there is a `pull_request` event. There are many `pull_requests` events, one of these is `syncronize` which happens to be triggered when there is a push to a branch that has an open PR  (stemming from the `on: push`
- it triggers builds to branches that do not have corresponding PRs. Unless I'm missing something I don't think we need that.

More on this here:
https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662/3
https://github.community/t/what-is-a-pull-request-synchronize-event/14784

What I think we might want is triggering the workflow on PRs being opened or updated, and on the main branch pushes (which I don't think we use but might be useful having anyway). So I'd go with something along these lines:
```suggestion
on:
  pull_request:
  push:
    branches: [ $default-branch ]
```
Note: the `$default-branch` is a built in var to reference the current default branch for the repository.

Reply:
This is a good point. It is currently redundant to do both in all cases. To be honest I was just copying https://github.com/pusher/pusher-http-php/pull/262. We should update both.

Reply:
What I'm not sure about is whether the merge commit to the master branch will be built too. [This thread] seems to imply that the `pull_request` `closed` event type will trigger this, but I suppose we won't know until we try...

Reply:
By default `pull_request`  `closed` is not a triggering event:


_"By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened. To trigger workflows for more activity types, use the types keyword."_